slack_sdk/oauth/authorize_url_generator/__init__.py

```diff
@@ -30,3 +30,33 @@ def generate(self, state: str):
         if self.redirect_uri is not None:
             url += f"&redirect_uri={self.redirect_uri}"
         return url
+
+
+class OpenIDConnectAuthorizeUrlGenerator:
+    """Refer to https://openid.net/specs/openid-connect-core-1_0.html"""
+    def __init__(
+        self,
+        *,
+        client_id: str,
+        redirect_uri: str,
+        scopes: Optional[Sequence[str]] = None,
+        authorization_url: str = "https://slack.com/openid/connect/authorize",
+    ):
+        self.client_id = client_id
+        self.redirect_uri = redirect_uri
+        self.scopes = scopes
+        self.authorization_url = authorization_url
+
+    def generate(self, state: str, nonce: Optional[str] = None):
+        scopes = ",".join(self.scopes) if self.scopes else ""
+        url = (
+            f"{self.authorization_url}?"
+            "response_type=code&"
+            f"state={state}&"
+            f"client_id={self.client_id}&"
+            f"scope={scopes}&"
+            f"redirect_uri={self.redirect_uri}"
+        )
+        if nonce is not None:
```

Comment:
nonce parameter is an optional one in the OpenID Connect 1.0 Specification https://openid.net/specs/openid-connect-core-1_0.html#AuthRequest